### PR TITLE
Fix psalm not running

### PR DESCRIPTION
**Pull request by CarlSchwan**
The issue was that we were using psalm/phar instead of vimeo/psalm. This
caused issue with the custom psalm plugin in build/psalm not using the same
php class anymore since HumbugBox was used as prefixed for the class inside
the phar.

This is using the opportunity to also update the psalm version from 3.8
to 3.18 and the php-cs-fixer too. This also consequentially updates the baseline.
There is, unfortunately, some recent regression for psalm that most of them seem
to be caused by recent commits and also a few false positives from psalm.

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

File: vendor-bin/psalm/composer.json [hunk 1]
```diff
@@ -1,10 +1,11 @@
 {
-	"config": {
```
Comment (skjnldsv) [inline vendor-bin/psalm/composer.json:2]: why removing the platform?

**Comment by CarlSchwan** [comment]
:( 

`./vendor-bin/psalm/vendor/vimeo/psalm/psalm --update-baseline` Works perfectly

`composer psalm:update-baseline` generate the broken baseline and doesn't report errors

I looked and both are using the same version and `composer psalm:update-baseline` should just call the binary inside lib/composer/bin that calls `./vendor-bin/psalm/vendor/vimeo/psalm/psalm` internally. But it seems there is something broken in the way this is done...

File: apps/encryption/lib/Recovery.php [hunk 2]
```diff
@@ -76,7 +76,7 @@ public function __construct(IUserSession $userSession,
 								IConfig $config,
 								IFile $file,
 								View $view) {
-		$this->user = ($userSession && $userSession->isLoggedIn()) ? $userSession->getUser() : false;
+		$this->user = ($userSession->isLoggedIn()) ? $userSession->getUser() : false;
```
Comment (come-nc) [inline apps/encryption/lib/Recovery.php:79]:
```suggestion
		$this->user = ($userSession->isLoggedIn()) ? $userSession->getUser() : null;
```

**Review by come-nc — APPROVED** [review]

**Comment by CarlSchwan** [comment]
Works now locally :) Let's hope the ci is also happy

**Comment by CarlSchwan** [comment]
The fact that `Code scanning results / Psalm` is failing is expected since it compares the baselines from the master and not with the baseline stored in the repo. There are due to the update to psalm 3.18 some new issues and also some issues that disappeared. There is nothing I can do about this, but once merged this failing CI should fix itself.

File: core/Command/Config/System/GetConfig.php [hunk 3]
```diff
@@ -65,9 +65,9 @@ protected function configure() {
 	 *
 	 * @param InputInterface  $input  An InputInterface instance
 	 * @param OutputInterface $output An OutputInterface instance
-	 * @return null|int null or 0 if everything went fine, or an error code
+	 * @return int 0 if everything went fine, or an error code
 	 */
-	protected function execute(InputInterface $input, OutputInterface $output): int {
+	protected function execute(InputInterface $input, OutputInterface $output) {
```
Comment (nickvergessen) [inline core/Command/Config/System/GetConfig.php:70]: ?

**Review by nickvergessen — APPROVED** [review]
Good apart from the one return

File: core/Command/Config/System/GetConfig.php (hunk 3 above)
Comment (CarlSchwan) [reply core/Command/Config/System/GetConfig.php]: Fixed

**Comment by skjnldsv** [comment]
/backport 47c9c34cca22602fb609de6c0e1d32ec17a9e7fe to stable22

**Comment by skjnldsv** [comment]
/backport 47c9c34cca22602fb609de6c0e1d32ec17a9e7fe to stable23

**Pull request merged**